crates/valence_protocol/src/packets/play/level_particles_s2c.rs

```diff
@@ -189,410 +230,86 @@ impl<'a> Particle {
+            Particle::Trail { .. } => 48,
+            Particle::ItemSlime => 49,
+            Particle::ItemCobweb => 50,
+            Particle::ItemSnowball => 51,
+            Particle::LargeSmoke => 52,
+            Particle::Lava => 53,
+            Particle::Mycelium => 54,
+            Particle::Note => 55,
+            Particle::Poof => 56,
+            Particle::Portal => 57,
+            Particle::Rain => 58,
+            Particle::Smoke => 59,
+            Particle::WhiteSmoke => 60,
+            Particle::Sneeze => 61,
+            Particle::Spit => 62,
+            Particle::SquidInk => 63,
+            Particle::SweepAttack => 64,
+            Particle::TotemOfUndying => 65,
+            Particle::Underwater => 66,
+            Particle::Splash => 67,
+            Particle::Witch => 68,
+            Particle::BubblePop => 69,
+            Particle::CurrentDown => 70,
+            Particle::BubbleColumnUp => 71,
+            Particle::Nautilus => 72,
+            Particle::Dolphin => 73,
+            Particle::CampfireCosySmoke => 74,
+            Particle::CampfireSignalSmoke => 75,
+            Particle::DrippingHoney => 76,
+            Particle::FallingHoney => 77,
+            Particle::LandingHoney => 78,
+            Particle::FallingNectar => 79,
+            Particle::FallingSporeBlossom => 80,
+            Particle::Ash => 81,
+            Particle::CrimsonSpore => 82,
+            Particle::WarpedSpore => 83,
+            Particle::SporeBlossomAir => 84,
+            Particle::DrippingObsidianTear => 85,
+            Particle::FallingObsidianTear => 86,
+            Particle::LandingObsidianTear => 87,
+            Particle::ReversePortal => 88,
+            Particle::WhiteAsh => 89,
+            Particle::SmallFlame => 90,
+            Particle::Snowflake => 91,
+            Particle::DrippingDripstoneLava => 92,
+            Particle::FallingDripstoneLava => 93,
+            Particle::DrippingDripstoneWater => 94,
+            Particle::FallingDripstoneWater => 95,
+            Particle::GlowSquidInk => 96,
+            Particle::Glow => 97,
+            Particle::WaxOn => 98,
+            Particle::WaxOff => 99,
+            Particle::ElectricSpark => 100,
+            Particle::Scrape => 101,
+            Particle::Shriek { .. } => 102,
+            Particle::EggCrack => 103,
+            Particle::DustPlume => 104,
+            Particle::TrialSpawnerDetection => 105,
+            Particle::TrialSpawnerDetectionOminous => 106,
+            Particle::VaultConnection => 107,
+            Particle::DustPillar(_) => 108,
+            Particle::OminousSpawning => 109,
+            Particle::RaidOmen => 110,
+            Particle::TrialOmen => 111,
+            Particle::BlockCrumble(_) => 112,
+            Particle::Firefly => 113,
         }
     }
-
-    /// Decodes the particle assuming the given particle ID.
-    pub fn decode_with_id(particle_id: i32, r: &mut &'a [u8]) -> anyhow::Result<Self> {
```

Comment:
There is no packet where we ever need to de/encode it without ID, therefore deriving Encode + Decode is enough